.cz.toml

```
@@ -0,0 +1,20 @@
+[tool.commitizen]
+name = "cz_customize"
+
+[tool.commitizen.customize]
+message_template = "{{scope}}: {{message}}"
+example = "flake: add new input"
+schema = "<scope>: <message>"
+schema_pattern = "^[a-zA-Z]+(?:\\/[a-zA-Z]+)?:\\s.+$"
+commit_parser = "^(?P<scope>[a-z]+(?:\\/[a-z]+)?):\\s(?P<message>.*)"
```

```suggestion
commit_parser = "^(?P<scope>[a-z]+(?:\\/[a-z]+)?):\\s(?P<message>\w.+)"
```

To ensure that the commit message actually does not contain punctuation at the start (and for it to not be empty).

---

Completely forgot to reply to this. `\w` doesn't seem to be parsed by commitizen:
```console
Failed to parse .cz.toml: Invalid character 'w' in string at line 9 col 66
```
When trying to commit with these lines. Could it be that it doesn't support this regex flavor?

---

Oh right, you need to escape the \

---

Ah apparently commit_parser is only there for changelog generation and not for validation, I you'd need to tweak `schema_pattern`, i.e 
```patch
diff --git a/.cz.toml b/.cz.toml
index 6015ff2..5f5a3d2 100644
--- a/.cz.toml
+++ b/.cz.toml
@@ -5,7 +5,7 @@ name = "cz_customize"
 message_template = "{{top_level_scope}}{% if specific_scope %}/{{specific_scope}}{% endif %}: {{message}}"
 example = "meta: add new input"
 schema = "<scope>: <message>"
-schema_pattern = "^[a-z]+(?:\\/[a-zA-Z0-9_-]+)?:\\s\\w.+$"
+schema_pattern = "^(programs|misc|lib|meta|docs|flakes)(?:\\/[a-zA-Z0-9_-]+)?:\\s\\w.+$"
 commit_parser = "^(?P<scope>[a-z]+(?:\\/[a-zA-Z0-9_-]+)?):\\s(?P<message>\\w.+)"
 
 [[tool.commitizen.customize.questions]]

```

(with added validation of tl scope)

but it's not really important